### PR TITLE
Only build index for OCP versioned apps

### PR DESCRIPTION
**Pull request by creydr**
Follow up on https://github.com/openshift-knative/hack/pull/334#issuecomment-2410846444

### CODE REVIEW

File: pkg/prowgen/prowgen_konflux.go [hunk 1]
```diff
@@ -295,7 +288,6 @@ func GenerateKonfluxServerlessOperator(ctx context.Context, openshiftRelease Rep
 			},
 			Excludes:       b.Konflux.Excludes,
 			ExcludesImages: b.Konflux.ExcludesImages,
-			FBCImages:      b.Konflux.FBCImages,
```
Comment (pierDipi) [inline pkg/prowgen/prowgen_konflux.go:298]: I would leave it here

Comment (pierDipi) [reply pkg/prowgen/prowgen_konflux.go:298]: in https://github.com/openshift-knative/hack/pull/341, it looks like we need it

Comment (creydr) [reply pkg/prowgen/prowgen_konflux.go:298]: I removed it only from the [`prowgen.Konflux`](https://github.com/openshift-knative/hack/blob/295c5f17bcbc79d3d0035737f6f4abb467bc70b8/pkg/prowgen/prowgen_config.go#L81) struct, but kept it in the [`konfluxgen.Config`](https://github.com/openshift-knative/hack/blob/295c5f17bcbc79d3d0035737f6f4abb467bc70b8/pkg/konfluxgen/konfluxgen.go#L59) struct.

**Pull request merged**